web/client-api/src/main/java/io/deephaven/web/client/api/subscription/TableViewportSubscription.java

```diff
@@ -109,9 +109,13 @@ public TableViewportSubscription(double firstRow, double lastRow, Column[] colum
         this.original = existingTable;
         this.originalState = original.state();
         copy = existingTable.copy(false).then(table -> new Promise<>((resolve, reject) -> {
-
             // Wait until the state is running to copy it
             originalState.onRunning(newState -> {
+                if (this.status == Status.DONE) {
+                    JsLog.debug("TableViewportSubscription closed before originalState.onRunning completed, ignoring");
+                    table.close();
```

Comment:
i dont think you need to call this - in the case where this.status == DONE (assigned in internalClose(), we have already closed this.table?

Reply:
Then instead of closing here, we should reject the promise.

Reply:
Nope, this doesn't work after all, table.close() probably makes the most sense.